lib/ddtrace/contrib/configuration/resolvers/regexp_resolver.rb

```diff
@@ -0,0 +1,31 @@
+require 'ddtrace/contrib/configuration/resolver'
+
+module Datadog
+  module Contrib
+    module Configuration
+      # Resolves a value to a configuration key
+      module Resolvers
+        # Matches strings against Regexps.
+        class RegexpResolver < Datadog::Contrib::Configuration::Resolver
+          def add_key(pattern)
+            patterns << pattern.is_a?(Regexp) ? pattern : /#{Regexp.quote(pattern)}/
+          end
+
+          def resolve(name)
+            return :default if name == :default
+
+            name = name.to_s
+            matching_pattern = patterns.find { |p| p =~ name }
```

Comment:
Maybe this isn't a big deal, but should we change `name` to `name.to_s`? Coercion might prevent some type incompatibility issues such as with `URI`, etc.